### DISCUSSION TITLE
Freeplane can't find Java runtime

### DISCUSSION THREAD

**Discussion by hobson42** [freeplane/freeplane#2284] [Help]
**Describe the bug**
When I click the freeplane icon, the program does not start, nor does any error message appear. 
When I type freeplane into a terminal I get - see below. 

**To Reproduce**
1. Open terminal
2. Type freeplane
The result is 
ian@cyllene:~$ freeplane
[warning] /usr/bin/freeplane: No java runtime was found
[error] /usr/bin/freeplane: Unable to find an appropriate java runtime. See java_wrappers(7) for help

**Expected behavior**
Find a runtime and continue to start. 

Clearly  apt install has not checked there was a suitable runtime. 

**Mind map files**
Can't get that far!

**Screenshots**

**Desktop (please complete the following information):**
 - OS:  Ubuntu 24.04 
 - 
 - Java Version OpenJDK Runtime Environment (build 21.0.3+9-Ubuntu-1ubuntu1)
 - 

**Additional context**
The fuller log. 
ian@cyllene:~$ sudo update-alternatives --config java
[sudo] password for ian: 
There are 2 choices for the alternative java (providing /usr/bin/java).

  Selection    Path                                         Priority   Status
------------------------------------------------------------
  0            /usr/lib/jvm/jdk-21-oracle-x64/bin/java       352346112 auto mode
  1            /usr/lib/jvm/java-21-openjdk-amd64/bin/java   2111      manual mode
* 2            /usr/lib/jvm/jdk-21-oracle-x64/bin/java       352346112 manual mode

Press <enter> to keep the current choice[*], or type selection number: 1
update-alternatives: using /usr/lib/jvm/java-21-openjdk-amd64/bin/java to provide /usr/bin/java (java) in manual mode
ian@cyllene:~$ java --version
openjdk 21.0.3 2024-04-16
OpenJDK Runtime Environment (build 21.0.3+9-Ubuntu-1ubuntu1)
OpenJDK 64-Bit Server VM (build 21.0.3+9-Ubuntu-1ubuntu1, mixed mode, sharing)

I have purged freeplane and reinstalled. 
I have found java_wrappers - but the files made no sense to me.

**Comment by dpolivaev** [comment]
The entire logic Freeplane uses to find Java on linux is contained in freeplane.sh. As a linux user please investigate what happens and suggest a fix. At the moment I do not have any linux system installed.

**Comment by OM-BDF** [comment]
Hello community,
I am wondering if someone did find a solution for this trouble?

I have exactly the same problem on a Linux Mint 22.1 system. 
I installed Freeplane via apt but it cannot find the JRE. I checked, the JRE and JDK are botth installed.

Any help is welcomed.
Thank you in advance.
Regards

**Reply by DaveC49** [reply]
On Linux MInt xena (22.3) I had to load Openjdk17 instead of Openjdk21 to get Freeplane to run.

**Comment by walmar** [comment]
Hi OM-BDF,
I'm not a daily Linux desktop user.

However, it seems to me that your request lacks basic background information, such as:
* The Freeplane version you try to run.
* If you have followed [Select Java installation used to run Freeplane - Command-line options and configuration - Freeplane Documentation](https://docs.freeplane.org/getting-started/Command-line_options_and_configuration.html#select-java-installation-used-to-run-freeplane).
* The output of running ``freeplane.sh`` in the terminal, as mentioned by dpolivaev and provided for #1792.

I doubt you'll get more attention unless you put in more effort yourself.

**Comment by OM-BDF** [comment]
Dear walmar,

sorry for having bothering you. How do you know the amount of effort I did put in myself before posting a help message?  I am a bit more than a newbie in the Linux world, but still. If that is a problem for you, please ignore me. If you like to help but need more data, just guide me.

Now, first information you listed, I wanted the latest Freeplane. 

In fact, when I did ask if someone found a solution to this trouble, it is because everything is described in the first message of the thread above.

So, in order to detail a bit more the amount of efforts I put it, I installed it via the GUI software installation app. The icon appears in the menu, but nothing was happening. I removed it using the GUI interface, took my breath and did install it via the command _apt install freeplane_ .
I cannot tell you which one has been installed as it never launched. There was no error screen neither. Nothing.
I also tried to launch it via the Terminal. The only output was that Java was not found.
I run _java --version_, got an answer. I also install the latest JDK, downloaded from the Java website. Removed freeplane and installed it again. Still no success.


Secondly, if I did post the message, it is because I did not find the two links you did provide here and that I came accross this issue thread.
**Thank to you, other people will be able to solve the same problem.**

On my end, as I encounter too many troubles for my work with Mint 22 (Freeplane was not the only software that was not running properly), I decided to scrap everything and went to Debian. I cannot anymore say if these advises will work on Mint or not.
It is a shame as I really did enjoy the feel of Mint. I might go back in the futur when I will have more time for solving this kind of troubles.

Regards

**Reply by walmar** [reply]
@OM-BDF

> How do you know the amount of effort I did put in myself before posting a help message?

I estimate it and in your case it seems to be coherent.

> I am a bit more than a newbie in the Linux world, but still. If that is a problem for you, please ignore me. If you like to help but need more data, just guide me.

I could indeed ignore the request, but at the other side, your are not the maintainer, so I don't have to.

If you are new to Linux I recommend you study the documents. These are usually accurately written by volunteers to educate users. I don't see the point in instructing everyone individually.

If you don't like the way I present my reply, then it might be better not to receive any answers.
I prefer to hold it short and crisp. It's not supposed to be as pleasant as being presented by a paid holiday mascot from an amusement park.

But yeah, I'm not going to respond to a possible next freestyle interpretation of the existing documentation, as there are presumably numerous other volunteers just waiting to help, perhaps by giving you instructions like "try reinstalling the operating system" that I won't interrupt.

> I wanted the latest Freeplane.

This is most likely not the version you will get this way according to #2119.

> In fact, when I did ask if someone found a solution to this trouble, it is because everything is described in the first message of the thread above.

No it is not, unless you continue to ignore the maintainer's comment.

> I cannot tell you which one has been installed as it never launched. There was no error screen neither. Nothing.

The package manager itself should be able to display the version of the installed packages.

> I also tried to launch it via the Terminal. The only output was that Java was not found.

``freeplane`` is not to be confused with ``freeplane.sh``. These are two different files.
The second one is not a binary file and does not contain the warning message mentioned, as you can verify yourself by browsing the [contents of this shell script](https://github.com/freeplane/freeplane/blob/1.12.x/freeplane_framework/script/freeplane.sh).

> On my end, as I encounter too many troubles for my work with Mint 22 (Freeplane was not the only software that was not running properly), I decided to scrap everything and went to Debian.

The behavior of Freeplane that you describe, that it does not work out of the box with the package manager of this Linux distribution, seems to be some kind of revelation of a false implication.

Given its history and my own past experience with it, Debian seems like a solid choice to me, as it is probably the distribution that has been best tested for stability.

At least you haven't moved on to the next more ambitious Linux distribution. This way, it may actually be more suited to your personal needs in the future.

**Comment by ghost** [comment]
@hobson42 @OM-BDF 

Are you using the Ubuntu/Debian/Mint etc binaries?
If so, they are somewhat outdated since the freeplane.sh dependency is on Java =< 17 version.

If you still want to use the old version, I suggest changing the freeplane.sh script from "java8" (or whatever) to "java17" and installing the java17 package with:
```
apt-get install openjdk-17-jdk openjdk-17-jre
```
But the best thing to do is to completely remove the package from the installed distribution and use the latest one available at:
("https://sourceforge.net/projects/freeplane/files/freeplane%20stable/")

If you have any knowledge about this, you can always help the Debian community to package the latest version of Freeplane in their repos (https://tracker.debian.org/pkg/freeplane). Debian need maintainers.

**Comment by marcinpraczko** [comment]
I have the same issue: 
```
$ lsb_release -rd 
No LSB modules are available.
Description:    Ubuntu 24.04.2 LTS
Release:        24.04
```

And `freeplane` is not starting - the same error like reported here:
- I've raised issue in ubuntu as well: https://bugs.launchpad.net/ubuntu/+source/freeplane/+bug/2109323

**Reply by Maltstar** [reply]
I got an issue on Ubuntu but I found out a solution maybe it might help you, see [issue #2444 ](https://github.com/freeplane/freeplane/discussions/2444)

**Reply by marcinpraczko** [reply]
I've run this as well - however this required to install older version of java :( 
Not ideal of modern systems - install old version (with security vulnerabilities) software to run another software.
But worked.
Solution is presented here: https://bugs.launchpad.net/ubuntu/+source/freeplane/+bug/2034752/comments/4

**Reply by marcinpraczko** [reply]
Hi @Maltstar  - thank you for your comment. I checked your solution and looks similar like this which I found. Both of us installed previous version of Java, but you adjusted shell script, my solution allowed to modify environment variable for running freeplane. 

Thank you for sharing your solution.

**Reply by dpolivaev** [reply]
Until now all old LTS Java versions are getting free security updates, and you can find them e.g. on page https://www.azul.com/downloads/?package=jdk#zulu. Even in the case they are not automatically integrated with OS package managers, you can easily get them. Personally I use https://sdkman.io/ for it, and there are many other solutions too.

**Reply by taekyunk** [reply]
The Freeplane version 1.7.10-1 is too old at this point and I have the same problem because it is the default version available in Software manager in Linux Mint 22.2

The solution for me was to download a newer version from the github.
1) I downloaded the most recent release version of 'deb' from here: https://github.com/freeplane/freeplane/releases/tag/release-1.12.14 and install it
2) Set the environment variable as `export FREEPLANE_JAVA_HOME=/usr/lib/jvm/java-21-openjdk-amd64/` in `.bashrc` 
This solved the problem for me.

**Reply by OM-BDF** [reply]
Dear taekyunk,
and all people having taken some time to help me.
After a long busy time, I was able to come back to this matter.
The Mint distro has been causing too much troubles for Freeplane but also severral other softwares I needed for my work. 
I therefore switched to Debian. 
I followed taekyunk's advises and can confirm that it is working fine for me.
Thank you all for your help.
Best regards,

**Reply by chairuou** [reply]
> Dear taekyunk, and all people having taken some time to help me. After a long busy time, I was able to come back to this matter. The Mint distro has been causing too much troubles for Freeplane but also severral other softwares I needed for my work. I therefore switched to Debian. I followed taekyunk's advises and can confirm that it is working fine for me. Thank you all for your help. Best regards,

@OM-BDF : I am on Mint 22.3 (zena) and here what's I did.

remove freeplace version installed from Mint repo 
**check java version** 
```
$ java --version
openjdk 21.0.10 2026-01-20
OpenJDK Runtime Environment (build 21.0.10+7-Ubuntu-124.04)
OpenJDK 64-Bit Server VM (build 21.0.10+7-Ubuntu-124.04, mixed mode, sharing)
```
**Download the latest release from Freeplane Github repo**

I am using this https://github.com/freeplane/freeplane/releases/download/release-1.13.2/freeplane_1.13.2.upstream-1_all.deb
then install it from local via `apt`
```
$ sudo apt deb ~/Downloads/freeplane_1.13.2.upstream-1_all.deb
$ freeplane
```
And it's working.
Final think: You needs create menu item then you can start it from menu with mouse.

<img width="1735" height="914" alt="image" src="https://github.com/user-attachments/assets/f3e2f2f8-d6f5-4486-b4fc-a16298dcc60b" />

**Comment by DaveC49** [comment]
Have just installed freepalne 1.13.2 as per these instructions on Linux
MInt 22.3 (Zena) and it is working fine also.

On Tue, 2026-05-26 at 21:27 -0700, chairuou wrote:
> > Dear taekyunk, and all people having taken some time to help me.
> > After a long busy time, I was able to come back to this matter. The
> > Mint distro has been causing too much troubles for Freeplane but
> > also severral other softwares I needed for my work. I therefore
> > switched to Debian. I followed taekyunk's advises and can confirm
> > that it is working fine for me. Thank you all for your help. Best
> > regards,
> > Dear taekyunk, and all people having taken some time to help me.
> > After a long busy time, I was able to come back to this matter. The
> > Mint distro has been causing too much troubles for Freeplane but
> > also severral other softwares I needed for my work. I therefore
> > switched to Debian. I followed taekyunk's advises and can confirm
> > that it is working fine for me. Thank you all for your help. Best
> > regards,
> @OM-BDF [1] : I am on Mint 22.3 (zena) and here what's I did.
> remove freeplace version installed from Mint repo
> check java version
> $ java --version
> openjdk 21.0.10 2026-01-20
> OpenJDK Runtime Environment (build 21.0.10+7-Ubuntu-124.04)
> OpenJDK 64-Bit Server VM (build 21.0.10+7-Ubuntu-124.04, mixed mode, sharing)
> Download the latest release from Freeplane Github repo
> I am using this
> https://github.com/freeplane/freeplane/releases/download/release-1.13.2/freeplane_1.13.2.upstream-1_all.deb
> then install it from local via apt
> $ sudo apt deb ~/Downloads/freeplane_1.13.2.upstream-1_all.deb
> $ freeplane
> And it's working.
> image.png (view on web) [2]
> —
> Reply to this email directly, view it on GitHub [3], or unsubscribe
> [4].
> Triage notifications on the go with GitHub Mobile for iOS [5] or
> Android [6].
> You are receiving this because you commented.Message ID:
> <freeplane/freeplane/repo-
> ***@***.***>

-- 
David Cousens

[1] @OM-BDF https://github.com/OM-BDF
[2] image.png (view on web)
    https://github.com/user-attachments/assets/f3e2f2f8-d6f5-4486-b4fc-a16298dcc60b
[3] view it on GitHub
    https://github.com/freeplane/freeplane/discussions/2284?email_source=notifications&email_token=ABW4CA4ID5LMBUT7FZHXZ6T44ZVCVA5CNFSNUABIM5UWIORPF5TWS5BNNB2WEL2ENFZWG5LTONUW63SDN5WW2ZLOOQXTCNZQG4YDMNBYUZZGKYLTN5XKOY3PNVWWK3TUUVSXMZLOOSWGM33PORSXEX3DNRUWG2Y#discussioncomment-17070648
[4] unsubscribe
    https://github.com/notifications/unsubscribe-auth/ABW4CA72MI5EQF5IBH7U67D44ZVCVAVCNFSM6AAAAACZFMOSNWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTOMBXGA3DIOA
[5] iOS
    https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
[6] Android
    https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub

**Reply by 5472qaywsx** [reply]
As of now, on my Debian testing machine, java is at version 25, which breaks compatibility with freeplane 1.13.2: can anybody confirm this?
Please let me know if you need more informations!

**Reply by macmarrum** [reply]
You can have a look at
- https://github.com/freeplane/freeplane/discussions/2124
